src/components/CreateEmployeeForm.tsx

```diff
@@ -1,128 +1,184 @@
-import React, { FC } from 'react';
+import React, { FC, useState } from 'react';
 import { useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
 import { yupResolver } from '@hookform/resolvers/yup';
 import * as yup from 'yup';
 
 import DropdownMenu from './DropdownMenu';
 import InputField from './InputField';
 import Label from './Label';
 import Button from './Button';
+import FileInput from './FileInput';
+import { useAddEmployeeMutation, useAddFileMutation, useGetDepartmentListQuery, useGetRoleListQuery }
+    from 'services/api';
+import PopUp from './PopUp';
 
 const schema = yup.object({
-    employeeName: yup.string().required('Employee Name is a required field'),
-    userName: yup.string().required('User Name is a required field'),
+    name: yup.string().required('Employee Name is a required field'),
+    username: yup.string().required('User Name is a required field'),
     age: yup.number().max(99, 'Enter a valid age').min(18, 'Enter a valid age')
-    .required().typeError('Age is a required field'),
+        .required().typeError('Age is a required field'),
     street: yup.string().required('Street is a required field'),
     city: yup.string().required('City is a required field'),
     state: yup.string().required('State is a required field '),
-    role: yup.string().required(),
-    department: yup.string().required(),
-    status: yup.string().required(),
-
+    email: yup.string().email('Not a valid e-mail id').required('E-mail is a required field'),
+    role_id: yup.number().required().typeError('Role is a required field '),
+    department_id: yup.number()
+        .required().typeError('Department is a required field '),
 });
 
 const CreateEmployeeForm: FC = () => {
-    const { register, handleSubmit, reset, formState: { errors } } = useForm(
+
+    const [errorMessage, setErrorMessage]=useState(false);
+
+    const { data: DepartmentData } = useGetDepartmentListQuery();
```

Comment:
Please refrain from making api calls in the component. Instead do the api calls from the parent containers and pass the data to the child component as prop. Please ensure the logic is always kept in containers, hooks or utils.